collect_app/src/main/java/org/odk/collect/android/utilities/CompressionUtils.java

```diff
@@ -66,13 +66,11 @@ public static String compress(String data) throws IOException {
     }
 
     public static String decompress(String compressedString) throws IOException, DataFormatException {
-        if (compressedString == null || compressedString.length() == 0) {
-            return compressedString;
+        if (compressedString == null || compressedString.length() == 0 || !compressedString.matches("^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$")) {
```

Comment:
It's not immediately clear what the regex is doing. Please make sure to include a comment.